pkg/ssh/ssh.go

```diff
@@ -119,13 +129,20 @@ func (s *Client) starting(ctx context.Context) error {
 		for {
 			cmd := exec.CommandContext(ctx, s.SSHCmd, flags...)
 
+			exitStatusWatcher := newExitStatusWatcher(os.Stderr)
 			cmd.Stdout = os.Stdout
-			cmd.Stderr = os.Stderr
+			cmd.Stderr = exitStatusWatcher
 			_ = cmd.Run()
 			if ctx.Err() != nil {
 				break // context was canceled
 			}
 
+			// The server won't accept new connections using the current tunnel id, there's no need to try again.
+			if exitStatusWatcher.exitStatus == ConnectionLimitReachedCode {
```

Comment:
Can't use the command exit code because it is not the exit code the pdc server sent

Reply:
It would be better if we didn't have to depend on stderr especially since the log statement we care about is at the debug level 🤔

Reply:
just poking around the docs a bit: I see [exec.Cmd](https://pkg.go.dev/os/exec#Cmd) has a `Process` property, which is of type `os.Process`. You can get an [`os.ProcessState`](https://pkg.go.dev/os#ProcessState) from this which has an `ExitSignal` property. Could we get the actual ssh exit signal from this?

Reply:
Will check

Reply:
That doesn't seem to work. I didn't find an ExitSignal property and the the exit code is 17

Reply:
Changed the exit code to 254